caas/kubernetes/provider/constraints.go

```diff
@@ -23,5 +27,81 @@ var unsupportedConstraints = []string{
 func (k *kubernetesClient) ConstraintsValidator(ctx envcontext.ProviderCallContext) (constraints.Validator, error) {
 	validator := constraints.NewValidator()
 	validator.RegisterUnsupported(unsupportedConstraints)
-	return validator, nil
+	return &constraintsValidator{Validator: validator}, nil
+}
+
+type constraintsValidator struct {
+	constraints.Validator
+}
+
+// Validate returns an error if the given constraints are not valid, and also
+// any unsupported attributes.
+func (v *constraintsValidator) Validate(cons constraints.Value) ([]string, error) {
+	if cons.Tags != nil {
+
+		var validateTopologyKeyTagFunc = func(key string, topologySpreadKey string, array []string) error {
+			var topologySpreadValue, otherValue *string = nil, nil
+			// Search for the topologySpreadValue for topologyKeyTag
+			for tag := range array {
+				if strings.HasPrefix(array[tag], topologySpreadKey) {
+					trimmedValue := strings.TrimPrefix(array[tag], topologySpreadKey+"=")
+					topologySpreadValue = &trimmedValue
+					break
+				}
+			}
+			if topologySpreadValue == nil {
+				// Nothing to do, continue to the next check
+				return nil
+			}
+
+			for tag := range array {
+				if strings.HasPrefix(array[tag], key) {
+					trimmedValue := strings.TrimPrefix(array[tag], key+"=")
+					otherValue = &trimmedValue
+					break
+				}
+			}
+
+			if otherValue != nil && *topologySpreadValue == *otherValue {
+				return errors.New("podAffinity/antiPodAffinity's topology-key and topologySpread's cannot have the same value")
+			}
+			return nil
+		}
+
+		positives, negatives := parseDelimitedValues(*cons.Tags)
+		elements := append(positives, negatives...)
+
+		topologySpreadKey := caasApplication.TopologySpreadPrefix + caasApplication.TopologyKeyTag
+		podKey := caasApplication.PodPrefix + caasApplication.TopologyKeyTag
+		if err := validateTopologyKeyTagFunc(podKey, topologySpreadKey, elements); err != nil {
+			return nil, err
+		}
+
+		antiPodKey := caasApplication.AntiPodPrefix + caasApplication.TopologyKeyTag
+		if err := validateTopologyKeyTagFunc(antiPodKey, topologySpreadKey, elements); err != nil {
+			return nil, err
+		}
+	}
+
+	return v.Validator.Validate(cons)
+}
+
+// parseDelimitedValues parses a slice of raw values coming from constraints
+// (Tags or Spaces). The result is split into two slices - positives and
```

Comment:
Spaces isn't relevant here. If code has been copied, comment needs to be tweaked.